lib/plugins/wallet/cardano/wallet.js

```diff
@@ -0,0 +1,76 @@
+const fs = require('fs')
+const { promisify } = require('util')
+
+const cardanoApi = require('./client')
+const crypto = require('crypto')
+
+const coinUtils = require('../../../coin-utils')
+const options = require('../../../options')
+
+const cryptoRec = coinUtils.getCryptoCurrency('ADA')
+const configPath = coinUtils.configPath(cryptoRec)
+
+function sha256(input) {
+  return crypto
+    .createHash('sha256')
+    .update(input)
+    .digest('base64')
+}
+
+async function getCardanoMnemonic() {
+  return promisify(fs.readFile)(options.cardanoMnemonicPath, 'utf8')
+    .then(text => text.trim())
+    .then(mnemonicText => ({
+      mnemonic: mnemonicText.split(/[ \r\n]+/),
+      hash: sha256(mnemonicText)
+    }))
+}
+
+async function getWalletConfiguration() {
+  return promisify(fs.readFile)(configPath)
+    .then(text => JSON.parse(text))
+    .catch(() => ({}))
+}
+
+async function saveWalletConfiguration(configuration) {
+  return fs.writeFile(configPath, JSON.stringify(configuration))
```

Comment:
uhuh, you are doing `promisify(fs.readFile)` but not `fs.writeFile`? Seems suspicious ...

Reply:
yes, because i suppose `async` make this for me - but i will test this and in case of failury i will change this

Reply:
`async` will not do automatic promisify imho